### PR TITLE
Add overloads for property types except enum

### LINKED ISSUE #70: Add overloads for each property type
source.createBoolProperty, createIntProperty, et al.

### PR DESCRIPTION
**Pull request by virtuald**
- Fixes #70

Tested with robotpy-cscore... didn't test nor compile Java, hopefully Travis does that for me. 👍

### CODE REVIEW

File: include/cscore_oo.h [hunk 1]
```diff
@@ -395,7 +395,7 @@ class CvSource : public VideoSource {
   /// @param description Description
   void SetDescription(llvm::StringRef description);
 
-  /// Create a property.
+  /// Create a property (deprecated)
```
Comment (PeterJohnson) [inline include/cscore_oo.h]: If we're going to mark this deprecated, we should do the same for the Java side.  I'd prefer to not deprecate it though.

Comment (virtuald) [reply include/cscore_oo.h]: It's just a nasty function.

Comment (virtuald) [reply include/cscore_oo.h]: But, ok, no deprecation.

**Comment by virtuald** [comment]
Updated.

**Review by PeterJohnson — APPROVED** [review]

**Pull request merged**